src/mame/machine/nb1412m2.cpp

```diff
@@ -314,13 +317,22 @@ void nb1412m2_device::dac_address_w(offs_t offset, uint8_t data)
 	}
 }
 
+void nb1412m2_device::dac_control_w(uint8_t data)
+{
+	if (data == 0)
+	{
+		// Mighty Guy is use this to stop psycho gun sound.
+		m_dac_playback = false;
+	}
+}
+
 // seems directly correlated to the DAC timer frequency
 // 0xd0 - 0xe0 - 0xf0 are the settings used
 void nb1412m2_device::dac_timer_w(uint8_t data)
 {
-//  popmessage("%02x",data);
-	// TODO: unknown algo, 0xe0*18 gives 4032 Hz which seems close enough for sample 36
-	m_dac_frequency = data*18;
+	// TODO: Argo is unknown
+	m_timer_rate = (data >> 4) - 11;
+	m_dac_frequency = m_timer_rate * 71 * 18;
```

Comment:
I'd guard here against dangerous negative DAC setup. 
If I have to guess an algo, bits 6 and 7 enables DAC interface and timer while bits 4 and 5 is frequency setting. 
Scale then is `N << ((data & 0x30) >> 4);` where N in your calcs makes 639. It is imho more likely to be rounded up to 640 Hz.

Reply:
Changed to be appropriate.
Result frequencies are now 1280, 2560, 3840 instead of 1278, 2556, 3834.